rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_chw_loops_purcahsed_cooling.py

```diff
@@ -0,0 +1,51 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    find_exactly_one_fluid_loop,
+    find_exactly_one_terminal_unit,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+EXTERNAL_FLUID_SOURCE = schema_enums["ExternalFluidSourceOptions"]
+FLUID_LOOP_TYPE = schema_enums["FluidLoopOptions"]
+
+
+def are_all_terminal_chw_loops_purchased_cooling(rmi_b, terminal_unit_id_list):
+    """Returns TRUE if the fluid loop associated with the cooling_from_loop associated with each terminal unit is purchased CHW. Returns FALSE if this is not the case.
+    ----------
+    rmi_b : json
+        RMD at RuleSetModelInstance level
+    terminal_unit_id_list : list
+        List of terminal units IDs
+    Returns
+    -------
+    bool
+        True: the fluid loop associated with the cooling_from_loop associated with each terminal unit is purchased CHW.
+        False: otherwise
+    """
+    are_all_terminal_chw_loops_purchased_cooling_flag = True
+
+    # get the list of loop ids if the external fluid source matches to either hot water or steam
+    purchased_cooling_loop_id_list_b = [
+        *find_all(
```

Comment:
Since `find_all` already returns a list, this can be simplified to
```
purchased_cooling_loop_id_list_b = 
    find_all(
        f'external_fluid_source[*][?(@.type="{EXTERNAL_FLUID_SOURCE.CHILLED_WATER}")].loop',
        rmi_b,
    )
```